src/librustdoc/json/conversions.rs

```diff
@@ -252,8 +252,11 @@ fn from_clean_item(item: clean::Item, tcx: TyCtxt<'_>) -> ItemEnum {
             bounds: b.into_iter().map(|x| x.into_tcx(tcx)).collect(),
             default: None,
         },
-        // FIXME: do not map to Typedef but to a custom variant
-        AssocTypeItem(t, _) => ItemEnum::Typedef(t.into_tcx(tcx)),
+        AssocTypeItem(t, b) => ItemEnum::AssocType {
```

Comment:
My nightly regression test job for [`cargo public-api`](https://github.com/Enselic/cargo-public-api) found a regression with this change ([job](https://github.com/Enselic/cargo-public-api/runs/7105199668?check_suite_focus=true)) 

Reported as https://github.com/rust-lang/rust/issues/98658